awx/api/views.py

```diff
@@ -2732,13 +2730,39 @@ def post(self, request, *args, **kwargs):
             if fd not in request.data and id_fd in request.data:
                 request.data[fd] = request.data[id_fd]
 
-        if get_request_version(self.request) == 1 and 'extra_credentials' in request.data:  # TODO: remove in 3.3
+        # This block causes `extra_credentials` to _always_ be ignored for
+        # the launch endpoint if we're accessing `/api/v1/`
+        if get_request_version(self.request) == 1 and 'extra_credentials' in request.data:
             if hasattr(request.data, '_mutable') and not request.data._mutable:
                 request.data._mutable = True
             extra_creds = request.data.pop('extra_credentials', None)
             if extra_creds is not None:
                 ignored_fields['extra_credentials'] = extra_creds
 
+        # Automatically convert legacy launch credential arguments into a list of `.credentials`
+        if 'credentials' not in request.data and (
+            'credential' in request.data or
+            'vault_credential' in request.data or
+            'extra_credentials' in request.data
```

Comment:
My preference would be to throw validation errors if user provides both "credentials" and any of these other ones (since "credentials" is new, it's not a breaking change).

I see a particular condition to put "extra_credentials" in `ignored_fields`, but not in a way that addresses the case of it being ignored because user also provided "credentials", using v2 API.